### DISCUSSION TITLE
Issues with swatch color ordering. Please help!

### DISCUSSION THREAD

**Discussion by eliduke** [mdbassit/Coloris#143] [Q&A] [answered]
Hey y'all! I am very new to Coloris, absolutely LOVING IT so far, but I have a couple questions that I can't seem to find answers to otherwheres. Sometimes, however, my eye balls are pretty selective and don't see things that right in front of them. That being said!

# Swatch Ordering Issues

To start, the swatch colors are loading up in the way I would expect. Things are a bit out of order, or so it seems.

For example, here's my javascript code:

```javascript
Coloris({
  el: ".color-field",
  theme: 'polaroid',
  themeMode: 'light',
  alpha: false,
  swatches: [
    "#f1463f",
    "#e62a65",
    "#9936ac",
    "#6344b2",
    "#3c57bo",
    "#1f9aee",
    "#odacef",
    "#1cbcd2",
    "#199588",
    "#54ac58",
    "#8fc057",
    "#cfd952",
    "#ffe858",
    "#ffbf3b",
    "#fd9731",
    "#fc5934"
  ]
});
```

I've got 16 colors going into the swatches. You can compare those colors with this visual list...

<img width="150" alt="swatch" src="https://github.com/mdbassit/Coloris/assets/1372520/98f15a8a-e346-4787-9254-31a236a7efc8">

But then this is what comes out...

<img width="286" alt="swatch2" src="https://github.com/mdbassit/Coloris/assets/1372520/c4525b8d-deca-4a76-bdba-05a6bfa19f88">

It looks pretty close to the colors from my swatch (there are indeed 16 colors there) and going left to right, top to bottom, I would think the first 6 colors in my list would be the top row in the color picker, right? It's close, but you can see in the top row that the 5th color over is not the right one. It should be #3C57B0, but it's not, and also that same wrong color is doubled up in the first spot on the next row. So the mystery color has taken the spots in the swatch where colors number 5 and number 7 should be.

And then it gets weirder!! With each click of a color in the swatch, it seems to randomly change those two duplicate colors, keeping them the same each time, but swapping in other random colors, sometimes setting them both to the color you just selected. AND THEN, on top of all that, if you ever click on either of the duplicate colors, it sets the color to #000000.

[swatch movie](https://github.com/mdbassit/Coloris/assets/1372520/ffc25f40-d8f6-476b-a4cf-5fce6682f2a5)

Something seems fishy! Thoughts? Any idea what's going on?

Thanks for your help! So glad that this tool is out there in the world with an engaged community around.

**Comment by eliduke** [comment]
I've got a small update. I decided to add a bigger swatch, with many more colors, and I'm getting many more duplicate colors...


[swatch movie](https://github.com/mdbassit/Coloris/assets/1372520/0245d802-4805-4f76-9292-2f1298a0a133)

**Reply by melloware** [reply]
Hmm this definitely feels like a bug to me.   Let's see what the boss says...

**Reply by eliduke** [reply]
Anyone else having this issue? Can it be reproduced elsewhere?

**Reply by mdbassit** [reply]
Yeah, that looks like a bug indeed. I will investigate.

**Reply by mdbassit** [reply, marked as answer]
It turns out it's not a bug, some color values in the swatches have the letter "o" instead of a zero. 
For instance `#3c57bo` should be `#3c57b0`. The perceived duplicates are a result of that invalid color values.

**Reply by eliduke** [reply]
WOAH! That's wild. Some truly wacky copy pasta right there. Thanks for helping me get to the bottom of it!
